Assets/Scripts/Game/CombatManager.cs

```diff
@@ -61,10 +93,16 @@ public bool CheckForEndCombat(GameObject _enemyToRemove)
             {
                 //Algorithm can be added here to roll for additional enemy reinforcements
 
+                //Exception for Vault Room
+                if (currentCarriage._isSpecialCarriage == CarriageData.SpecialCarriageExceptions.Vault && isBossAlive)
+                {
+                    SpawnEnemies();
+                    return false;
+                }
+
                 EndCombat();
                 return true;
             }
-
             else return false;
```

Comment:
I'm generally just unsure how to restructure the code to be cleaner. So I added comments to describe why each return statement exists, and what they mean.

But otherwise, code is intact as is. Sorry!